### PR TITLE
add the option to render vendor extensions

### PR DESCRIPTION
**Pull request by ingoclaro**
implements #522

### CODE REVIEW

File: src/components/Fields/Extensions.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import { SchemaModel } from '../../services/models';
+import { FieldDetail } from './FieldDetail';
+
+export interface ExtensionsProps {
+  schema: SchemaModel;
+}
+
+export class Extensions extends React.PureComponent<ExtensionsProps> {
+  redocExtensions = [
```
Comment (RomanGotsiy) [inline src/components/Fields/Extensions.tsx]: Could you move this array to https://github.com/Rebilly/ReDoc/blob/master/src/utils/openapi.ts

Also, having it as an object would be a bit faster than `array.includes` lookup.

File: src/components/Fields/Extensions.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    'x-nullable',
+    'x-servers',
+    'x-tagGroups',
+    'x-traitTag',
+  ];
+
+  render() {
+    const { schema } = this.props;
+    const fullSchema = schema.schema;
+    const extensionList = Object.keys(fullSchema).filter(
```
Comment (RomanGotsiy) [inline src/components/Fields/Extensions.tsx]: Also, I think there is sense to do these filterings only when `options.showExtensions` is true.

I would add method called `getExtension`:

```ts
getExtensions() {
  const { schema } = this.props;
  const fullSchema = schema.schema;
  return Object.keys(fullSchema).filter(
    key => key.startsWith('x-') && !this.redocExtensions.includes(key),
  );
}
```

and then just used it in the `render` like bellow:

```ts
render() {
  return (
    <OptionsContext.Consumer>
      {options => (
        <>
          {options.showExtensions &&
            this.getExtensions().map(key => (
              <FieldDetail key={key} label={key} value={fullSchema[key]} />
            ))}
        </>
      )}
    </OptionsContext.Consumer>
  );
}
```

**Comment by ingoclaro** [comment]
thanks for the feedback, I refactored the code.

**Comment by ingoclaro** [comment]
@RomanGotsiy ping

**Comment by RomanGotsiy** [comment]
@ingoclaro sorry for the delay!
Super busy...

Thanks for your efforts!

I will review it ASAP. Most likely at the beginning of the next week.

Please, ping me again if I disappear 🙈

**Comment by ingoclaro** [comment]
resolved conflicts. ping @RomanGotsiy

**Comment by RomanGotsiy** [comment]
I pushed a commit on top of this PR:
- implemented filtering by name (i.e. you can provide array of extensions names to show)
- added extensions to Operation
- extensions on `Parameter` object are now also included, not only on `Schema` Object
- changed look of extensions:
image

@ingoclaro Could you check it out and let me know what do you think?

**Comment by ingoclaro** [comment]
@RomanGotsiy looks good to me. Thanks for the improvement.

**Comment by jeremyfiel** [comment]
@RomanHotsiy @tatomyr    how can I use this in redocly cli with `preview-docs`?   no flag exists on this command for `--showExtensions`

**Comment by tatomyr** [comment]
@AlexVarchuk  could you help with that?

**Comment by AlexVarchuk** [comment]
Hi @jeremyfiel, you can create a config file, and pass it to the command
```
redocly preview-docs <api> [--config=<path>] [--port=<value>] [--host=<host>]
```

**Comment by jeremyfiel** [comment]
What is the config attribute name?

**Pull request merged**